### ISSUE TITLE
Uyuni Bugreport - Uyuni 2021.12 - venv-salt-minion for EL 8 - dnfnotify plugin only distributed for python3.0

### ISSUE THREAD

**Issue by lumarel** [uyuni-project/uyuni#4545]
# Additional Information

It looks like the venv-salt-minion package only delivers the dnfnotify plugin for python3.9, but as RHEL 8 and clones use python3.6 as the default python3 version for the os it is unable to find the plugin and doesn't send the notifications.

## Version of Uyuni Server and Proxy (if used)
```bash
# dnf info Uyuni-Server-release
Last metadata expiration check: 2:22:40 ago on Thu 09 Dec 2021 01:25:14 PM CET.
Installed Packages
Name         : Uyuni-Server-release
Version      : 2021.11
Release      : 161.3.uyuni1
Architecture : x86_64
Size         : 1.4 k
Source       : Uyuni-Server-release-2021.11-161.3.uyuni1.src.rpm
Repository   : @System
From repo    : @commandline
Summary      : Uyuni Server
```

## Details about issue
```bash
# dnf info venv-salt-minion
Last metadata expiration check: 0:15:08 ago on Thu 09 Dec 2021 03:14:15 PM CET.
Installed Packages
Name         : venv-salt-minion
Version      : 3002.2
Release      : 37.1.uyuni
Architecture : x86_64
Size         : 90 M
Source       : venv-salt-minion-3002.2-37.1.uyuni.src.rpm
Repository   : @System
From repo    : susemanager:bootstrap
Summary      : The venvjailed client component for Salt
```

If any further information is needed, please just notify me, I'm happy to help!

**Comment by lumarel** [comment]
Also note, I noticed that the susemanager.py plugin is also only showing up in the `/usr/lib/python3.9/site-packages/dnf-plugins/` directory and not in the python3.6 directory, this is different to the "old" directly installed salt-minion, which makes them available in both folders. (this makes dnf unable to sync any packages or metadata from Uyuni's repos)
As this file is not part of the package I'm not sure how that gets distributed, but I suppose it is synced on the initial registration against Uyuni.

**Comment by lumarel** [comment]
@vzhestkov Sorry for pinging you, but this might be interesting for you, as we already talked about that in Gitter

**Comment by vzhestkov** [comment]
@lumarel sorry I missed the notification from github about this issue.
The issue appears due to wrong way of calculating the file path in the context of using salt bundle. Will fix it.

**Comment by sbluhm** [comment]
From previous discussion: CentOS 7 can run `dnf` and `yum` at the same time. Or just one of them. (I have both).
Uyuni bootstrap installed `yum-rhn-plugin.noarch` which hopefully has a dependency on `yum` so it would be safe to assume that yum is always on a salt minion.

**Comment by vzhestkov** [comment]
Not exactly. CentOS 8 has yum, but yum plugins are not invoked at all, at least yumnotify.

**Comment by sbluhm** [comment]
> @sbluhm in your case just removing the dnf package will solve highstate apply error.

@vzhestkov I can confirm that. Thank you. So there probably has to be a "Conflict: dnf" in one of the packages.
The workaround for my purpose would be to remove all dnf from the C7 machines and add an alias dnf=yum.

> Not exactly. CentOS 8 has yum, but yum plugins are not invoked at all, at least yumnotify.
If I remember correct, `yum` on CentOS 8 is just a dnf wrapper. That's why I had to rewrite the whole Spacewalk reposync for dnf. Not sure what `dnf` is on CentOS 7. Wouldn't be surprised if it is just a yum wrapper instead. (I might be completely wrong here that there are just version incompatibilities in there somewhere. Too long ago).

**Comment by vzhestkov** [comment]
@lumarel @sbluhm here is the PR with the fix for this issue: https://github.com/uyuni-project/uyuni/pull/4626

**Comment by sbluhm** [comment]
@vzhestkov , I can confirm this one working:
```

----------
          ID: mgrchannels_susemanagerplugin_dnf
    Function: file.managed
        Name: /usr/lib/python2.7/site-packages/dnf-plugins/susemanagerplugin.py
      Result: true
     Comment: File /usr/lib/python2.7/site-packages/dnf-plugins/susemanagerplugin.py is in the correct state
     Started: 19:29:45.490849
    Duration: 56.16
         SLS: channels
     Changed: {}
----------
          ID: mgrchannels_susemanagerplugin_conf_dnf
    Function: file.managed
        Name: /etc/dnf/plugins/susemanagerplugin.conf
      Result: true
     Comment: File /etc/dnf/plugins/susemanagerplugin.conf is in the correct state
     Started: 19:29:45.547160
    Duration: 16.656
         SLS: channels
     Changed: {}
----------

```

**Comment by vzhestkov** [comment]
Closing as fixed
@sbluhm thanks for confirming
